### PR TITLE
[Kernels] Improve Triton fp8 block scaled kernel

### PR DESCRIPTION
**Pull request by lgeiger**
## Purpose

This PR aims to improve performance of the Triton fp8 block scaled w8a8 kernel.

It's probably best to review it commit by commit:

- 8d1c78b07989d1bbbecffd9738aaa4c7dea18ba5 simplifies the pointer arithmetic
- 34154269f39e4078d2c8a3bbf6570ed4a12a48a0 removes the masked load for cases where `K % BLOCK_SIZE_K == 0`. To me this seems to always be the case, but I might miss some edge cases so I kept a fallback to the previous behaviour when this condition isn't met.
- 91ef1859ef71e0535e70d998c287fb16c25f755f re-tunes the triton configs on a L40s following #29217

## Test Plan

Correctness should be covered by `tests/kernels/quantization/test_block_fp8.py` and I also verified it with lm_eval for Qwen3-VL-2B-Instruct-FP8.

I tested performance on a L40s with `Qwen3-VL-32B-Instruct-FP8`:

```
vllm serve Qwen/Qwen3-VL-32B-Instruct-FP8 --limit-mm-per-prompt.video 0 --limit-mm-per-prompt.image 0 --max-model-len 24000 --no-enable-prefix-caching

vllm bench serve --backend vllm --model Qwen/Qwen3-VL-32B-Instruct-FP8 --endpoint /v1/completions --dataset-name sharegpt --dataset-path ShareGPT_V3_unfiltered_cleaned_split.json --num-prompts 1000
```

## Test Results

**Before:**

```
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  282.88
Total input tokens:                      217393
Total generated tokens:                  189963
Request throughput (req/s):              3.54
Output token throughput (tok/s):         671.54
Peak output token throughput (tok/s):    1477.00
Peak concurrent requests:                1000.00
Total Token throughput (tok/s):          1440.06
---------------Time to First Token----------------
Mean TTFT (ms):                          115438.24
Median TTFT (ms):                        119080.39
P99 TTFT (ms):                           239007.78
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          129.52
Median TPOT (ms):                        104.82
P99 TPOT (ms):                           540.94
---------------Inter-token Latency----------------
Mean ITL (ms):                           104.46
Median ITL (ms):                         66.16
P99 ITL (ms):                            438.74
==================================================
```

**After code changes:**

```
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  279.72
Total input tokens:                      217393
Total generated tokens:                  189963
Request throughput (req/s):              3.58
Output token throughput (tok/s):         679.12
Peak output token throughput (tok/s):    1707.00
Peak concurrent requests:                1000.00
Total Token throughput (tok/s):          1456.30
---------------Time to First Token----------------
Mean TTFT (ms):                          117722.64
Median TTFT (ms):                        122529.00
P99 TTFT (ms):                           247590.06
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          133.14
Median TPOT (ms):                        106.13
P99 TPOT (ms):                           503.34
---------------Inter-token Latency----------------
Mean ITL (ms):                           106.04
Median ITL (ms):                         66.58
P99 ITL (ms):                            491.64
==================================================
```

**After code changes and re-tuned:**

```
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  273.16
Total input tokens:                      217393
Total generated tokens:                  189963
Request throughput (req/s):              3.66
Output token throughput (tok/s):         695.43
Peak output token throughput (tok/s):    1644.00
Peak concurrent requests:                1000.00
Total Token throughput (tok/s):          1491.28
---------------Time to First Token----------------
Mean TTFT (ms):                          117644.34
Median TTFT (ms):                        123531.08
P99 TTFT (ms):                           240034.56
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          122.76
Median TPOT (ms):                        102.63
P99 TPOT (ms):                           511.67
---------------Inter-token Latency----------------
Mean ITL (ms):                           102.85
Median ITL (ms):                         71.09
P99 ITL (ms):                            518.81
==================================================
```

Overall this improves throughput of `Qwen3-VL-32B-Instruct-FP8` on a single L40s by 3.5% which is decent.

### CODE REVIEW

File: vllm/model_executor/layers/quantization/utils/fp8_utils.py [hunk 1]
```diff
@@ -714,20 +715,24 @@ def _w8a8_triton_block_scaled_mm(
...
+            a = tl.load(a_ptrs)
+            b = tl.load(b_ptrs)
+        a_s = tl.load(As_ptrs)
+        b_s = tl.load(Bs_ptrs)
 
         accumulator += tl.dot(a, b) * a_s[:, None] * b_s[None, :]
         a_ptrs += BLOCK_SIZE_K * stride_ak
         b_ptrs += BLOCK_SIZE_K * stride_bk
+        As_ptrs += scale_step_k * stride_As_k
+        Bs_ptrs += scale_step_k * stride_Bs_k
```
[the remark this answers is not in the document]
Comment (lgeiger) [reply vllm/model_executor/layers/quantization/utils/fp8_utils.py:735]: This sounds sensible. I have a look later
